src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -565,6 +578,40 @@ def resolve_imports_and_packages(
             max_batch_size,
             source_code_display=source_code_display,
         )
+
+        if pickled_by_reference:
+            try:
+                source_code = inspect.getsource(func)
+                with tempfile.TemporaryDirectory() as tempdir:
+                    abs_path = os.path.join(
+                        tempdir, f"{func.__module__.replace('.', '/')}.py"
+                    )
+                    os.makedirs(os.path.dirname(abs_path), exist_ok=True)
+                    with open(abs_path, "w") as fp:
+                        fp.write(source_code)
+                    additional_imports = {}
+                    resolved_import_tuple = session._resolve_import_path(
+                        abs_path, func.__module__
+                    )
+                    additional_imports[
+                        resolved_import_tuple[0]
+                    ] = resolved_import_tuple[1:]
+                    all_urls += session._resolve_imports(
+                        upload_stage,
+                        additional_imports,
+                        statement_params=statement_params,
+                    )
+            except TypeError as exc:
+                if (
+                    str(exc)
+                    != "module, class, method, function, traceback, frame, or code object was expected, got builtin_function_or_method"
```

Comment:
If the exception message is like this, then we do nothing? Where is TypeError raised? Let's add a comment here?